osutil/user.go

```diff
@@ -46,23 +55,38 @@ func AddExtraSudoUser(name string, sshKeys []string, gecos string) error {
 		"--gecos", gecos,
 		"--extrausers",
 		"--disabled-password",
-		"--add_extra_groups", "sudo",
 		name)
 	if output, err := cmd.CombinedOutput(); err != nil {
 		return fmt.Errorf("adduser failed with %s: %s", err, output)
 	}
 
+	// Must escape "." as files containing it are ignored in sudoers.d.
+	sudoersFile := filepath.Join(sudoersDotD, "create-user-"+strings.Replace(name, ".", "%2E", -1))
```

Comment:
This is the only change on top of @mwhudson's branch, besides adding tests for this which were lacking in the original.